lib/utils/enrich.js

```diff
@@ -0,0 +1,31 @@
+const prettyHrtime = require('pretty-hrtime');
+
+const log = require('../utils/logger');
+
+function enrich(ids, enrichmentData, description) {
+  const startTime = process.hrtime();
+  const enrichmentReportObj = {
+    baseObjectCount: ids.length,
+    enrichmentObjectCount: enrichmentData.length,
+    mismatchedObjectCount: 0,
+    mismatchedObjectIds: [],
+  };
+
+  const enrichedData = ids.map((idObj) => {
```

Comment:
It's more conventional to use a `reduce` when removing elements from an array, rather than to use a map which returns `undefined`, then filtering out `undefined` items, i.e. something like:
```
const enrichedData = ids.reduce((accumulator, idObj) => {
   ...
   if (matchedObj) {
      accumulator.push(newObj);
    } else {
      ...
   }
}, []); 
```